### PR TITLE
Move storybook

### LINKED ISSUE #7415: Setup Storybook in a different folder and package
# Description

Move current Storybook configuration to a separated folder and package, avoiding to install and uninstall while developing.

## Context

As Storybook requires several libraries to be installed, this has been avoided due process time. It is seems quite straightforward the possibility to have separated configurations in a similar configuration

```
📁 dashboard
├─ 📁 shell
|  └─ 📄 package.json
└─ 📁 storybook
   ├─ 📁 .storybook
   |  ├─ 📄 main.js
   |  └─ 📄 preview.js
   └─ 📄 package.json
```

Existing paths defined in the `.storybook` configuration, such like `main.js` and `preview.js`, will require to update existing paths, e.g.:

```
module.exports = {
-  stories: ["~shell/**/stories/**/*.stories.tsx"],
+  stories: ["../shell/**/stories/**/*.stories.tsx"],
}
```

It will also require to update scripts in the `./package.json` and related scripts for installation.
`.github/workflows/storybook` will also require a review.

### Notes

- Beware of [existing issues in CI](https://github.com/rancher/dashboard/actions/runs/3580734095/jobs/6023105668)
- Fix issues with current build randomly failing
- Remove [verbose log in CI](https://github.com/rancher/dashboard/actions/runs/3575747702/jobs/6012675841) while publishing

### PR DESCRIPTION
**Pull request by nwmac**
Fixes #7415 

This PR moves the `.storybook` and `stories` folders under a new top-level `storybook` folder.

The new `storybook` folder contains its own `package.json` with `nohoist` set - so Storybook dependencies will not interfere with the top-level `node_modules` folder.

You can now developer dashboard and storybook easily from a checkout.

Docs and build scripts have been updated.

Note: This PR includes updates from 2 other Storybook PRs that should be reviewed/merged first:  #7602 and #7601

### CODE REVIEW

**Review by cnotv — CHANGES_REQUESTED** [review]
Looks good, thank you for taking care of it. Only concern to me are the part related to the installation as you can read in the comment.

PS: Do not forget to rebase the incoming merges.

File: package.json [hunk 1]
```diff
@@ -44,11 +45,7 @@
     "e2e:pre-prod": "yarn docker:local:stop && yarn docker:local:start && DEV_PORTS=true yarn build",
     "e2e:prod": "START_SERVER_AND_TEST_INSECURE=1 server-test start:prod https-get://localhost:8005 cy:run:sorry",
     "coverage": "npx nyc merge coverage coverage/coverage.json",
-    "storybook": "yarn run install-storybook && start-storybook -p 6006",
-    "build-storybook": "yarn run install-storybook && build-storybook",
-    "install-storybook": "./scripts/storybook-install",
-    "remove-storybook": "./scripts/storybook-install -d",
-    "remove-storybook-deps": "./scripts/storybook-install -r",
+    "storybook": "cat storybook/moved.txt",
```
[the remark this answers is not in the document]
Comment (nwmac) [reply package.json:48]: I'm okay with that - done

[the remark this answers is not in the document]
Comment (nwmac) [reply package.json:48]: Done

**Comment by nwmac** [comment]
Should all be good now @cnotv

**Pull request merged**